distributed/preloading.py

```diff
@@ -3,25 +3,71 @@
 import os
 import shutil
 import sys
+import filecmp
 from importlib import import_module
 
+import click
+
 from .utils import import_file
 
 logger = logging.getLogger(__name__)
 
 
-def preload_modules(names, parameter=None, file_dir=None):
-    """ Imports modules, handles `dask_setup` and `dask_teardown` functions
+def validate_preload_argv(ctx, param, value):
+    """Click option callback providing validation of preload subcommand arguments."""
+    if not value and not ctx.params.get("preload", None):
+        # No preload argv provided and no preload modules specified.
+        return value
+
+    if value and not ctx.params.get("preload", None):
+        raise click.UsageError("Additional preload arguments specified without --preload target.")
```

Comment:
This might be valuable to include in a test.  We want to make ensure that operations like `dask-scheduler --bad-keyword` do genuinely raise informative arguments.  In these cases we probably don't want to talk about `--preload` but rather just mention that there are unknown keywords (and perhaps list what they are).

Reply:
Updated in commit below to match click error messages for unknown options and extra arguments.